src/test/java/games/strategy/engine/lobby/server/db/DbUserControllerTest.java

```diff
@@ -0,0 +1,227 @@
+package games.strategy.engine.lobby.server.db;
+
+import static org.hamcrest.CoreMatchers.nullValue;
+import static org.hamcrest.CoreMatchers.sameInstance;
+import static org.hamcrest.MatcherAssert.assertThat;
+
+import org.hamcrest.core.Is;
+
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.mockito.junit.MockitoJUnitRunner;
+
+import games.strategy.engine.lobby.server.userDB.DBUser;
+
+@RunWith(MockitoJUnitRunner.class)
+public class DbUserControllerTest {
+
+  private DbUserController testObj;
+
+  @Mock
+  private UserDao mockPrimary;
+
+  @Mock
+  private UserDao mockSecondary;
+
+  @Mock
+  private MigrationCounter mockMigrationCounter;
+
+  @Before
+  public void setup() {
+    testObj = new DbUserController(mockPrimary, mockSecondary, mockMigrationCounter);
+  }
+
+  /**
+   * Happiest case of login, success on the secondary (new DB), no fallback, we just return true and record stats.
+   */
+  @Test
+  public void login_SecondarySucceeds() {
```

Comment:
I was very surprised that it was in there. I noted it a few PRs ago when some underscores were used to split up test cases. The naming convention doc is pretty clear that there would no underscores, I do suppose they have it in there just for test cases. Agree it should be reserved for test cases.